### PR TITLE
Add documentation for localization

### LINKED ISSUE #56: Add Documentation for Localization into README
Can we please include a brief section in readme.md outlining the steps we would need to complete to add a new locale.

### PR DESCRIPTION
**Pull request by alicoding**
fixes #56

This is my first write up on localization. Hopefully I didn't miss
anything here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,3 +16,60 @@ $> cp sample.env .env
...
+
+`en-US.yaml`
+``` yaml
+---
+en-US:
+  first: This is your first message
+  second: This is your second message
+```
+
+You have to make sure you match your language code in your YAML file and the name of the file with what you include in your config file for the converting part otherwise it will fail.
```
Comment (ScottDowne) [inline README.md]: I think we can simplify this line, it's a bit wordy and hard to read.

Something like:

"The language code and title of the YAML file must match a locale in the `supportedLocales` intl-config.json array"

That is, if I understand what you were initially trying to say.

File: README.md [hunk 2]
```diff
@@ -16,3 +16,60 @@ $> cp sample.env .env
...
+  }
+
+});
+```
+
+Once you added the mixin it will expose `getIntlMessage` method for your component to get the localized message for the given key name.
+
+#### Adding locale
+
+Since we are using YAML for our translation, but React-Intl expect JSON which will require an extra build step to convert from YAML to JSON.
```
Comment (ScottDowne) [inline README.md]: Maybe better?

"Because we are using YAML for our translation and React-Intl expects JSON, we need an extra build step to convert YAML to JSON"

File: README.md [hunk 3]
```diff
@@ -16,3 +16,60 @@ $> cp sample.env .env
...
+#### Localize a component or page
+
+To localize a component or page you have to always include IntlMixin in your class mixins for example:
+
+``` typescript
+var React = require('react');
+var IntlMixin = require('react-intl').IntlMixin;
+
+var Example = React.createClass({
+	mixins: [IntlMixin]
```
Comment (ScottDowne) [inline README.md]: Some white space we can remove.

File: README.md [hunk 4]
```diff
@@ -16,3 +16,60 @@ $> cp sample.env .env
...
+      <div>
+      	<h1>{this.getIntlMessage('key_name_here')}
+      </div>
+    );
+  }
+
+});
+```
+
+Once you added the mixin it will expose `getIntlMessage` method for your component to get the localized message for the given key name.
```
Comment (ScottDowne) [inline README.md]: I think what `getIntlMessage` does is probably pretty clear. It's more important that we're very clear about how to load it, so I think the less we say here the clearer it is. So maybe something like this:

"Adding the `IntlMixin` mixin exposes `getIntlMessage` to your component."

File: README.md (hunk 1 above)
Comment (alicoding) [inline README.md]: pretty much except file name and locale in YAML must match what's in `supportedLocales`

`en-US.yaml` and in YAML `en-US` at the top.

File: README.md [hunk 5]
```diff
@@ -16,3 +16,60 @@ $> cp sample.env .env
...
 $> npm start
 ```
+
+## Localization
+
+In this project we're using [React-Intl](https://github.com/yahoo/react-intl) to localize our application and YAML for translation.
+
+#### Localize a component or page
+
+To localize a component or page you have to always include IntlMixin in your class mixins for example:
```
Comment (ScottDowne) [inline README.md]: Oh, I was reading things bottom up, and didn't see this before doing the comment below.

Comment (ScottDowne) [inline README.md]: Some small changes here:

"To localize a component or page you have to include `IntlMixin` in your class `mixins`, for example"

File: README.md (hunk 4 above)
Comment (ScottDowne) [inline README.md]: Reading this from the top down, I'm now mostly OK with this bit, I just have tiny changes now.

"Once you add the mixin it will expose `getIntlMessage` method to your component to get the localized message for the given key"

I changed added/add for/to and removed name

**Comment by ScottDowne** [comment]
I have a handful of comments that might make this clearer.

**Comment by alicoding** [comment]
@ScottDowne fixed all of them. Can you re-review again?

**Comment by ScottDowne** [comment]
Looks good. Once Travis is done merge it!

**Comment by ScottDowne** [comment]
Oh, Travis is done, so I shall merge.
